tools/chocolateyinstall.ps1

```diff
@@ -9,12 +9,32 @@ $url             = 'https://dl.influxdata.com/telegraf/releases/telegraf-1.4.0_w
 $url64           = 'https://dl.influxdata.com/telegraf/releases/telegraf-1.4.0_windows_amd64.zip'
 $fileLocation    = Join-Path $install_folder 'telegraf.exe'
 
+# Make sure the config directory exists
 If(!(Test-Path -Path $configDirectory)){
   New-Item -Path $configDirectory -ItemType Directory
 }
 
-If (Get-Service -Name "telegraf" -ErrorAction SilentlyContinue) {
+# If telegraf.exe exists, and the service is running, stop the service
+If (Test-Path -Path $fileLocation){
+  If (Get-Service -Name "telegraf" -ErrorAction SilentlyContinue) {
+    Stop-Service -Name "telegraf"
+    Start-Sleep -s 10
+  }
+}
+
+# If telegraf.exe exists, and the service is enabled, uninstall the service
+If (Test-Path -Path $fileLocation){
+  If (Get-Service -Name "telegraf" -ErrorAction SilentlyContinue) {
     & $fileLocation --service uninstall
+  }
+}
+
+# if the service is already defined, do not install the service
```

Comment:
I'm trying to understand how the binary got removed in your situation which caused the service not to get uninstalled? Neither the chocolateyinstall.ps1 nor the chocolateybeforemodify.ps1 have any logic to remove the binaries.

Reply:
chocolatey itself tries to replace the binary during the install process. I can send over the logs if you would like.